### DISCUSSION TITLE
Something went wrong while executing your query C7D2:390D:1412CC3:18B7FA1:5C4E105A

### DISCUSSION THREAD

**Discussion by givanse** [community/community#24408] [Apps, API and Webhooks]
<p>The query I’ve using has been working for a long while now and recently it stoped working.</p>
<p>The most relevant portion reads:</p>
<pre><code>This may be the result of a timeout, or it could be a GitHub bug.
Please include `C7D2:390D:1412CC3:18B7FA1:5C4E105A` when reporting this issue.
</code></pre>
<p>Full error message:</p>
<pre><code>Something went wrong while executing your query. This may be the result of a timeout, or it could be a GitHub bug. Please include `C7D2:390D:1412CC3:18B7FA1:5C4E105A` when reporting this issue.: {"response":{"data":"null","errors":[{"message":"Something went wrong while executing your query. This may be the result of a timeout, or it could be a GitHub bug. Please include `C7D2:390D:1412CC3:18B7FA1:5C4E105A` when reporting this issue."}],"status":502},"request":{"query":"\nquery ProjectPulse {\n\n org0: organization(login: \"ethereum\") {\n ...orgRepos\n }\n}\n\nfragment repoPulse on Repository {\n nameWithOwner,\n forkCount,\n isFork,\n parent {\n nameWithOwner\n },\n isMirror,\n mirrorUrl,\n createdAt,\n description,\n releases(last: 1, orderBy:{direction:ASC,field:CREATED_AT}) {\n last: nodes {\n name,\n updatedAt\n }\n },\n master: ref(qualifiedName: \"master\") {\n commits: target {\n ... on Commit {\n history {\n totalCount\n }\n }\n }\n },\n prsOpen: pullRequests(states:OPEN, last: 100) {\n totalCount\n },\n prsClosed: pullRequests(states:CLOSED, last: 100) {\n totalCount\n },\n issuesOpen: issues(states:OPEN, last: 100) {\n totalCount\n },\n issuesClosed: issues(states:CLOSED, last: 100) {\n totalCount\n },\n stargazers {\n totalCount\n },\n watchers {\n totalCount\n }\n}\n\nfragment orgRepos on Organization {\n repositories(last: 100, orderBy:{direction:ASC,field:UPDATED_AT}) {\n nodes {\n ...repoPulse\n }\n }\n}\n"}}
</code></pre>

**Comment by nickvanw** [comment, marked as answer]
<p>Hey <a class="mention" href="/givanse">@givanse</a>!</p>
<p>Do you have a copy of the query, by any chance? </p>
<p>Looking at the error on our end, it appears that your query timed out - this can happen when a user is requesting a significant amount of data, especially when maximizing their connections and asking for 100 resources at a time.</p>
<p>If possible, I would recommend cutting down on the number of resources you are requesting at once, and doing more overall queries. This will allow each one to fit into the allowed time, and hopefully result in more successful queries!</p>

**Reply by garylab** [reply]
I decreased `issues(first:100, ...)` to `issues(first:30, ...)`, it works!😊😊😊

**Comment by givanse** [comment]
<p>Thanks for confirming that it is a timeout!</p>
<p>After reading your reply I realized that:</p>
<p> - it is a dynamically generated query, and</p>
<p> - the number of repos/orgs (in my data set) must have increased enought to push it over the limit</p>
<p>case solved, I’ll split my query</p>

**Comment by AndreaGriffiths11** [comment]
<p>Thanks, <a class="mention" href="/givanse">@givanse</a> for coming back and providing feedback on what worked! We’re looking forward to seeing you around!</p>

**Comment by jverb093** [comment]
<p><a class="mention" href="/nickvanw">@nickvanw</a>  I’ve tried cutting down on the number of resources I’m requesting at once, and I’m still getting this error. To be fair, the repo I’m trying to pull from has ~150,000 pull requests. I see your other suggestion is to do more overall queries. How would I go about doing that? For reference, this is what my query looks like: </p>
<pre><code>query {
        repository(name: $name, owner: $owner) {
            pullRequests(first: 100, after:$cursor) {
                pageInfo {
                    hasNextPage
                    endCursor
                }
                edges {
                    node {
                        id
                        author {
                            login
                            }
                        title
                        number
                        createdAt
                        state
                        mergeable
                        mergedAt
                        merged
                        closed
                        closedAt
                        changedFiles
                        additions
                        deletions
                    }
                }
            }
        }
    }
</code></pre>
